### PR TITLE
[1928] Escape peroid in User email regex

### PR DESCRIPTION
**Pull request by richrace**
### Context

Noticed Regex did not escape any character `.`

### Changes proposed in this pull request

Escape any character regex 

### Guidance to review

Edge case spec added

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -80,12 +80,26 @@
...
-      it "does shows up in User.non_admins" do
-        expect(User.non_admins).to eq([subject])
+        its(:admin?) { should be_falsey }
+
+        it "does shows up in User.non_admins" do
+          expect(User.non_admins).to eq([subject])
+        end
+      end
+
+      context 'when other email looks like "digital.education.gov.uk" doamin' do
```
Comment (faissalMT) [inline spec/models/user_spec.rb]:
```suggestion
      context 'when an email has a domain similar but not identical to "digital.education.gov.uk"' do
```

File: spec/models/user_spec.rb [hunk 2]
```diff
@@ -80,12 +80,26 @@
...
+        let(:email) { 'test@digitalaeducation.gov.uk' }
+
+        its(:admin?) { should be_falsey }
+
+        it "does shows up in User.non_admins" do
```
Comment (faissalMT) [inline spec/models/user_spec.rb]: Try to describe tests from the perspective of the user
```suggestion
        it 'is not an admin' do
```

Comment (richrace) [reply spec/models/user_spec.rb]: Should this be changed in all other uses see: https://github.com/DFE-Digital/manage-courses-backend/blob/79da4a30a7dae58011e797c13ae726f66af99ff2/spec/models/user_spec.rb#L67 or https://github.com/DFE-Digital/manage-courses-backend/blob/79da4a30a7dae58011e797c13ae726f66af99ff2/spec/models/user_spec.rb#L77

File: app/models/user.rb [hunk 3]
```diff
@@ -18,7 +18,7 @@
 class User < ApplicationRecord
   include AASM
 
-  DFE_EMAIL_PATTERN = '@(digital.){0,1}education.gov.uk$'.freeze
+  DFE_EMAIL_PATTERN = '@(digital\.){0,1}education.gov.uk$'.freeze
```
Comment (timabell) [inline app/models/user.rb]: good spot

**Comment by richrace** [comment]
[Incorrect Regex for `DFE_EMAIL_PATTERN`](https://trello.com/c/NZPQ64sf/1928-incorrect-regex-for-dfeemailpattern)

File: app/models/user.rb (hunk 3 above)
Comment (richrace) [reply app/models/user.rb]: Just realised I forgot about the other two `.`. Updating now.

File: spec/models/user_spec.rb [hunk 4]
```diff
@@ -88,16 +88,34 @@
...
 
-      its(:admin?) { should be_falsey }
+      context 'when other doamin' do
+        let(:email) { 'test@hrmc.gov.uk' }
 
-      it "does shows up in User.non_admins" do
-        expect(User.non_admins).to eq([subject])
+        its(:admin?) { should be_falsey }
+
+        it "is an non-admin user" do
```
Comment (defong) [inline spec/models/user_spec.rb]:
```suggestion
        it "is a non-admin user" do
```

File: spec/models/user_spec.rb [hunk 5]
```diff
@@ -88,16 +88,34 @@
     end
 
     context 'user does not have a digital.education or education.gov.uk email' do
-      subject { create(:user, email: 'test@hrmc.gov.uk') }
+      subject { create(:user, email: email) }
 
-      its(:admin?) { should be_falsey }
+      context 'when other doamin' do
```
Comment (defong) [inline spec/models/user_spec.rb]:
```suggestion
      context 'when other domain' do
```

File: spec/models/user_spec.rb [hunk 6]
```diff
@@ -88,16 +88,34 @@
     end
 
     context 'user does not have a digital.education or education.gov.uk email' do
```
Comment (defong) [inline spec/models/user_spec.rb]:
```suggestion
    context 'user does not have a digital.education.gov.uk or education.gov.uk email' do
```

File: spec/models/user_spec.rb [hunk 7]
```diff
@@ -88,16 +88,34 @@
...
       end
 
-      it "doesn't show up in User.admins" do
-        expect(User.admins).to be_empty
+      context 'when an email has a domain similar but not identical to "digital.education.gov.uk"' do
+        let(:email) { 'test@digitalaeducationagov.uk' }
+
+        its(:admin?) { should be_falsey }
+
+        it "is an non-admin user" do
```
Comment (defong) [inline spec/models/user_spec.rb]:
```suggestion
        it "is a non-admin user" do
```

**Pull request merged**